### ISSUE TITLE
Rackspace Cloud Files Deploy Fails with: undefined local variable or method `__dir__'

### ISSUE THREAD

**Issue by carolynvs** [travis-ci/dpl#506]
I have a build that deploys tagged commits to Rackspace Cloud Files. It is now failing with a ruby error that I'm not sure how to either fix or workaround.

Failing Build: https://travis-ci.org/getcarina/carina/builds/160255161
Last Successful Build: https://travis-ci.org/getcarina/carina/builds/128882102

```
Installing deploy dependencies
[TRUNCATED]
22 gems installed

/home/travis/.rvm/gems/ruby-1.9.3-p551/gems/fog-profitbricks-2.0.1/lib/fog/profitbricks.rb:4:in `<top (required)>': undefined local variable or method `__dir__' for main:Object (NameError)
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/fog-1.34.0/lib/fog.rb:52:in `<top (required)>'
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:128:in `require'
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:128:in `rescue in require'
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:39:in `require'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/provider.rb:75:in `requires'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/provider/cloud_files.rb:9:in `<class:CloudFiles>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/provider/cloud_files.rb:5:in `<class:Provider>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/provider/cloud_files.rb:4:in `<module:DPL>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/provider/cloud_files.rb:3:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/provider.rb:57:in `const_get'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/provider.rb:57:in `block in new'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/cli.rb:41:in `fold'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/provider.rb:54:in `new'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/cli.rb:31:in `run'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/lib/dpl/cli.rb:7:in `run'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18/bin/dpl:5:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `load'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `<main>'
failed to deploy
```

This is my deploy configuration:

```yaml
deploy:
  - provider: cloudfiles
    username: ***
    api-key:
      secure: ***
    region: IAD
    container: carina-downloads
    skip_cleanup: true
    on:
      repo: getcarina/carina
      tags: true
```

**Comment by carolynvs** [comment]
Is there a way to change the version of ruby used during the deploy? Sounds like fog-profitbricks uses `__dir__` which isn't available: https://github.com/fog/fog-profitbricks/issues/24.

**Comment by carolynvs** [comment]
Here is a workaround that fixes the deployment

```yaml
before_deploy:
  - rvm 1.9.3 do gem install fog-profitbricks -v 0.0.5
```

**Comment by BanzaiMan** [comment]
This has been fixed in 1.8.19.

**Comment by carolynvs** [comment]
Thanks you!
